lib/sort_by_length.rb

```diff
@@ -1,7 +1,22 @@
 # A method which will return an array of the words in the string
 #  sorted by the length of the word.
-# Time complexity: ?
-# Space complexity: ?
+# Time complexity: O(n) -> n because iterates over the length of the array.
+# Space complexity: O(1)
+
 def sort_by_length(my_sentence)
```

Comment:
The time complexity here is O(n^2) because you have the two nested loops.  This is an implementation of BubbleSort.